motor_control/Triple_HBridge/3PhaseMotorControl_UCC277.ino

```diff
@@ -0,0 +1,219 @@
+        initMotor();
+        interrupting = false; 
+     }
+   }
+   
+
+   
+
+}
+
+
+
+
+// Could get rid of the read statements by writing three different codes that changes the value of "hall" based on which pin is the interrupt but not sure how much efficiency we'd really get out of that
+void State_Change_1(){
+
+      if(changing_state) // triggered if interrupt function is called during other function
+         interrupting = true;
+
+         
+      if (pin1_state == 0){
+
+        
+        analogWrite(outHighPins[0], motor_speed);
+        digitalWrite(outLowPins[0], LOW); 
+        
+        analogWrite(outHighPins[1], 0);
+        digitalWrite(outLowPins[1], HIGH);
+   
+      }
+
+      else{
+
+        analogWrite(outHighPins[0], 0);
+        digitalWrite(outLowPins[0], HIGH);
+
+        analogWrite(outHighPins[1], motor_speed);
+        digitalWrite(outLowPins[1], LOW);
+
+      }
+      
+       analogWrite(outHighPins[2], 0);
+       digitalWrite(outLowPins[2], LOW);
+
+       pin1_state != pin1_state;
+       interrupt_time = 0;
+      
+}
+
+
+
+void State_Change_2(){
+  
+      if(changing_state) // triggered if interrupt function is called during other function
+         interrupting = true;
+         
+      if (pin2_state == 0){
+
+        analogWrite(outHighPins[1], motor_speed);
+        digitalWrite(outLowPins[1], LOW);
+        
+        analogWrite(outHighPins[2], 0);
+        digitalWrite(outLowPins[2], HIGH);
+      }
+
+      else{
+
+        analogWrite(outHighPins[1], 0);
+        digitalWrite(outLowPins[1], HIGH);
+
+        analogWrite(outHighPins[2], motor_speed);
+        digitalWrite(outLowPins[2], HIGH);
+      }
+
+
+
+      
+       analogWrite(outHighPins[0], 0);
+       digitalWrite(outLowPins[0], LOW);
+
+       pin2_state != pin2_state;
+       interrupt_time = 0;
+      
+}
+
+void State_Change_3(){
+  
+       if(changing_state) // triggered if interrupt function is called during other function
+         interrupting = true;
+         
+      if (pin3_state == 0){
+        analogWrite(outHighPins[0], 0);
+        digitalWrite(outLowPins[0], HIGH);
+
+        analogWrite(outHighPins[2], motor_speed);
+        digitalWrite(outLowPins[2], LOW);
+
+
+      }
+
+      else{
+        analogWrite(outHighPins[0], motor_speed);
+        digitalWrite(outLowPins[0], LOW);
+
+        analogWrite(outHighPins[2], 0);
+        digitalWrite(outLowPins[2], HIGH);
+
+      }
+
+
+        analogWrite(outHighPins[1], 0);
+        digitalWrite(outLowPins[1], LOW);
+
+       pin3_state != pin3_state;
+       interrupt_time = 0;
+      
+}
+
+void initMotor(){
+  byte hallIndex = 4 * (byte)hallStates[0] + 2 * (byte)hallStates[1] + (byte)hallStates[2] - 1;
+
+  int powerToWrite[6][3] = {
+  {2,0,1}, // hallReadings 001
+  {0,1,2}, // hallReadings 010
+  {0,2,1}, // hallReadings 011
+  {1,2,0}, // hallReadings 100
+  {1,0,2}, // hallReadings 101
+  {2,1,0}  // hallReadings 110
+  };
+

```

Comment:
woops this part needs to be changed. Currently it is using the IR2104 motor driving properties (notice that it says shutPins rather than outLowPins). Very minimal change needed